code/game/objects/items/stacks/stack.dm

```diff
@@ -52,14 +52,16 @@
 	update_icon(UPDATE_ICON_STATE)
 
 /obj/item/stack/update_icon_state()
-	. = ..()
 	if(!dynamic_icon_state)
-		return
-	var/temp_amount = get_amount()
-	if(temp_amount > 1)
-		icon_state = "[initial(icon_state)]_[min(temp_amount, 3)]" //2 if amount is 2, 3 if more.
-		return
-	icon_state = initial(icon_state)
+		return ..()
+	if(amount <= (max_amount * (1/3)))
+		icon_state = initial(icon_state)
+		return ..()
+	if (amount <= (max_amount * (2/3)))
+		icon_state = "[initial(icon_state)]_2"
+		return ..()
+	icon_state = "[initial(icon_state)]_3"
+	return ..()
```

Comment:
This is a little bit messy, I think if we found the amount/max_amount, then made a special case for less than one third, it would be a lot cleaner.
```suggestion
    var/state = (amount/max_amount) * 3
	if(state <= 1)
		icon_state = initial(icon_state)
		return
	icon_state = "[initial(icon_state)]_[state]"
```